src/Symfony/Component/ObjectMapper/ObjectMapper.php

```diff
@@ -203,7 +207,7 @@ private function getSourceValue(object $source, object $target, mixed $value, \S
             } elseif ($objectMap->contains($value)) {
                 $value = $objectMap[$value];
             } else {
-                $value = $this->map($value, $mapTo->target);
+                $value = $this->objectMapper->map($value, $mapTo->target);
```

Comment:
let's save a circular reference:
```suggestion
                $value = ($this->objectMapper ?? $this)->map($value, $mapTo->target);
```